File: app/src/main/java/com/app/nikhil/coroutinedownloader/injection/scope/ViewModelKey.java
package com.app.nikhil.coroutinedownloader.injection.scope;

import androidx.lifecycle.ViewModel;
import dagger.MapKey;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@MapKey
public @interface ViewModelKey {
  Class<? extends ViewModel> value();
}

File: app/src/main/java/com/app/nikhil/coroutinedownloader/injection/scope/ViewModelKey.kt
package com.app.nikhil.coroutinedownloader.injection.scope

import androidx.lifecycle.ViewModel
import dagger.MapKey
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.FUNCTION
import kotlin.reflect.KClass

@Target(FUNCTION)
@Retention(RUNTIME)
@MapKey
annotation class ViewModelKey (val value: KClass<out ViewModel>)